src/torrent.cpp

```diff
@@ -8738,6 +8744,7 @@ namespace {
 	{
 		TORRENT_ASSERT(is_single_thread());
 		if (m_sequential_download == sd) return;
+		if (!sd) set_sequential_start(piece_index_t(0), piece_index_t(-1));
```

Comment:
I would expect the piece picker logic to be simpler if you have to set the end to one-past-end rather than -1. You wouldn't need any special cases (I think).